.ruff.toml

```diff
@@ -37,15 +40,19 @@ fixable = [
     "D", # pydocstyle
 ]
 
+src = ["src", "tests", "examples"]
+
 target-version = "py39"
 
 [mccabe]
 max-complexity = 10
 
 [per-file-ignores]
 "scripts/*" = ["PL", "S", "SIM", "D"]
-"tests/*" = ["S", "SIM"]
+"tests/*" = ["S", "SIM", "PLR"]
 ".devcontainer/*" = ["S", "SIM", "D"]
+"examples/*" = ["S", "D"]
+"__init__.py" = ["D"]
```

Comment:
Maybe you can be a bit more specific with these exclusion (exclude only certain rules, not the whole rule set).